File: src/main/java/by/kohanova/model/Friends.java
package by.kohanova.model;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "friends")
public class Friends implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Integer id;

    @ManyToOne
    @JoinColumn(name = "user_alfa")
    public User userAlfa;

    @ManyToOne
    @JoinColumn(name = "user_beta")
    public User userBeta;

    @Column
    public boolean status;

    @Override
    public boolean equals(Object obj) {
        if (obj == null)
            return false;
        if (!(obj instanceof Friends))
            return false;
        Friends other = (Friends) obj;
        return id != null && id.equals(other.id);
    }

    @Override
    public int hashCode() {
        return id == null ? 0 : id.hashCode();
    }
}

File: client/src/app/model/friends.ts
import {User} from "./user";
interface IFriends {
    id?: number;
    friend1?: User;
    friend2?: User;
    status?: boolean;
}
export class Friends implements IFriends {
    id: number;
    friend1: User;
    friend2: User;
    status: boolean;

    constructor(obj: IFriends = {}as IFriends) {
        let {
            friend1 = null,
            friend2 = null,
            status = false,

        }:IFriends = obj as IFriends;
        this.friend1 = friend1;
        this.friend2 = friend2;
        this.status = status;

    }
}